docs/topics/multiplatform-mobile/multiplatform-mobile-upgrade-app.md

```diff
@@ -286,89 +486,258 @@ is already imported and used in `ContentView.swift` with `import shared`.
+    kotlin.sourceSets.all {
+        languageSettings.optIn("kotlin.experimental.ExperimentalObjCName")
+    }
+    ```
+   
+##### Mark the flow with KMP-NativeCoroutines
+
+1. Navigate to the `shared/src/commonMain` directory and locate the `Greeting.kt` file.
+2. Add the `@NativeCoroutines`annotation to the `greet()` function. This will ensure that the plugin generates the right
+   code to support correct flow handling on iOS:
+  
+   ```kotlin
+    import com.rickclephas.kmp.nativecoroutines.NativeCoroutines
+    
+    class Greeting {
+        // ...
+       
+        @NativeCoroutines
+        fun greet(): Flow<String> = flow {
+            // ...
         }
     }
     ```
 
-    * Kotlin/Native [provides bidirectional interoperability with Objective-C](https://kotlinlang.org/docs/native-objc-interop.html#mappings), thus
-      Kotlin concepts, including `suspend` functions, are mapped to the corresponding Swift/Objective-C concepts and vice versa. When you
-      compile a Kotlin module into an Apple framework, suspending functions are available in it as functions with
-      callbacks (`completionHandler`).
-    * The `greet()` function was marked with the `@Throws(Exception::class)` annotation. So any exceptions that are
-      instances of the `Exception` class or its subclass will be propagated as `NSError`, so you can handle them in the `completionHandler`.
-    * When calling Kotlin `suspend` functions from Swift, completion handlers might be called on threads other than main,
-      see the [iOS integration](native-ios-integration.md#completion-handlers) in the Kotlin/Native memory manager.
-      That's why `DispatchQueue.main.async` is used to update `phrases` property.
-
-5. In `ContentView_Previews`, ensure that the view model is properly initialized:
-
-   ```Swift
-   struct ContentView_Previews: PreviewProvider {
-       static var previews: some View {
-           ContentView(viewModel: ContentView.ViewModel())
+##### Import the library using SPM in XCode
+
+1. In Xcode, right-click the `iosApp` project in the left-hand project menu and select **Add packages**.
+2. In the search bar, enter the package name:
+
+     ```none
+    https://github.com/rickclephas/KMP-NativeCoroutines.git
+    ```
+
+   ![Importing KMP-NativeCoroutines](multiplatform-import-kmp-nativecoroutines.png){width=700}
+
+3. Keep the default options, "Branch" for **Dependency Rule** and "master" for **Version Rule**, and click
+   the **Add Package** button.
+4. In the next window, select "KMPNativeCoroutinesAsync" and "KMPNativeCoroutinesCore" and click **Add Package**:
+
+   ![Add KMP-NativeCoroutines packages](multiplatform-add-package.png){width=500}
+
+    This will install the KMP-NativeCoroutines packages necessary to work with the `async/await` mechanism.
+
+##### Consume the flow using the KMP-NativeCoroutines library
+
+1. In `iosApp/ContentView.swift`, update the `startObserving()` function to consume the flow using KMP-NativeCoroutine's
```

Comment:
Maybe a code block here, to show how it should be done?

Reply:
I don't know, it ends up the same block twice

Reply:
My point was that you ask a reader to do something but don't provide the code block. 
For me, it is a bit more difficult to understand what you are asking to do with your code just with text, without any code.